File: Solutions/IQCare.Core/IQCare.PMTCT.Core/Models/HEI/Vaccination.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace IQCare.PMTCT.Core.Models.HEI
{
    public class Vaccination
    {
        public int Id { get; set; }
        public int PatientId { get; set; }
        public int PatientMasterVisitId { get; set; }
        public int Vaccine { get; set; }
        public string VaccineStage { get; set; }
        public bool DeleteFlag { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? VaccineDate { get; set; }
        public bool Active { get; set; }
        public int? AppointmentId { get; set; }
        public int? PeriodId { get; set; }
        [NotMapped]
        public DateTime? NextSchedule { get; set; }
    }
}

File: Solutions/IQCare.Core/IQCare/frontend/app/pmtct/_models/hei/Vaccination.ts
export interface Vaccination {
    Id?: number;
    PatientId?: number;
    PatientMasterVisitId?: number;
    Vaccine?: number;
    VaccineStage?: number;
    DeleteFlag?: number;
    CreatedBy?: number;
    CreateDate?: Date;
    VaccineDate?: Date;
    Active?: number;
    AppointmentId?: number;
    Period?: number;
    NextSchedule?: Date;
}
